crates/settings/src/settings_file.rs

```diff
@@ -115,7 +115,19 @@ pub fn update_settings_file<T: Settings>(
         let new_text = cx.read_global(|store: &SettingsStore, _cx| {
             store.new_text_for_update::<T>(old_text, update)
         })?;
-        fs.atomic_write(paths::SETTINGS.clone(), new_text).await?;
+        let initial_path = paths::SETTINGS.as_path();
+        let resolved_path = fs
+            .canonicalize(initial_path)
+            .await
+            .with_context(|| format!("Failed to canonicalize settings path {:?}", initial_path))?;
+        fs.atomic_write(resolved_path.clone(), new_text)
+            .await
+            .with_context(|| {
+                format!(
+                    "Failed to write settings to file {:}",
```

Comment:
Sorry, one last thing here: 
```suggestion
                    "Failed to write settings to file {:?}",
```

and no need to call `display()` since Debug impl will place quotes around the path (separating it from the rest of the log which might be useful if the path will have spaces in it) and print the whole path.

I would have fixed it myself, but GH interface does not allow to push into the contributors' branches, alas.

Reply:
ah, yep, that was me not noticing missing the `?` and taking a compiler suggestion to fix it 😃